browser-extension/mv3/src/content-scripts/client/responseRuleHandler.ts

```diff
@@ -1,9 +1,15 @@
 import { getEnabledRules, onRuleOrGroupChange } from "common/rulesStore";
 import { ResponseRulePair, RuleType } from "common/types";
-import { executeDynamicScriptOnPage } from "../utils";
+import { executeDynamicScriptOnPage, isExtensionEnabled } from "../utils";
 import { PUBLIC_NAMESPACE } from "common/constants";
 
 const cacheResponseRules = async () => {
```

Comment:
@nafees87n The caching logic is getting duplicated right now. We should move this into a separate function and reuse it possibly like this

```
const cacheRulesOnPage = async(ruleType) => {
  const isExtensionEnabled = await isExtensionEnabled();

  if (!isExtensionEnabled) return;

  const rules = await getEnabledRules(ruleType);
  if (!rules.length) return;

  let ruleTypeVariableExposedOnWindow = null;
  if (ruleType === RuleType.RESPONSE) ruleTypeVariableExposedOnWindow = 'responseRules'
  ....

  
  executeScriptOnPage(
  `
    window.${PUBLIC_NAMESPACE}....
    window.${PUBLIC_NAMESPACE}.${ruleTypeVariableExposedOnWindow} = ....
  `
  )
  
}

```